ksqldb-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -392,6 +399,10 @@ public SchemaKGroupedStream groupBy(
     );
   }
 
+  protected List<SqlType> toSqlTypes(List<Expression> expressions) {
+    return expressions.stream().map(typeManager::getExpressionSqlType).collect(Collectors.toList());
```

Comment:
Hmm.. I'm wondering if we do need to re-parse the expression again to get the sql types (I filed https://github.com/confluentinc/ksql/issues/7066 before feeling that we are doing multiple duplicated passes today). If we have to, and this is not on the critical paths, maybe it's fine. cc @agavra @AlanConfluent could you share your thoughts?

Reply:
Not on the CC list but my two cents anyway 😄 : I think this is fine since, as you pointed out, this is not on the critical path. This code path is only hit once while validating, starting, or re-starting the query.

Reply:
+1 to @vcrfxia's comment - it might be worth it in the future but not at the expense of a bigger PR here 😉